src/Models/QuickFormModel.js

```diff
@@ -0,0 +1,184 @@
+/* Copyright (c) 2014, SalesLogix, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @class Sage.Platform.Mobile.Models._ModelBase
+ * Model is the base class for all data models.
+ *
+ * 
+ * @alternateClassName _ModelBase
+
+ */
+define('Mobile/SalesLogix/Models/QuickFormModel', [
+    'dojo/_base/declare',
+    'dojo/_base/lang',
+    'Sage/Platform/Mobile/Models/_ModelBase',
+    'Mobile/SalesLogix/Models/QuickFormControls/TextControl',
+    'Mobile/SalesLogix/Models/QuickFormControls/AddressControl',
+    'Mobile/SalesLogix/Models/QuickFormControls/EmailControl',
+    'Mobile/SalesLogix/Models/QuickFormControls/PhoneControl',
+    'Mobile/SalesLogix/Models/QuickFormControls/NameControl',
+    'Mobile/SalesLogix/Models/QuickFormControls/DateControl',
+    'Mobile/SalesLogix/Models/QuickFormControls/BooleanControl',
+    'Mobile/SalesLogix/Models/QuickFormControls/UrlControl',
+    'Mobile/SalesLogix/Models/QuickFormControls/OwnerControl',
+    'Mobile/SalesLogix/Models/QuickFormControls/UserControl',
+    'Mobile/SalesLogix/Models/QuickFormControls/PicklistControl',
+    'Mobile/SalesLogix/Models/QuickFormControls/LookupControl',
+    'Mobile/SalesLogix/Models/QuickFormControls/NumericControl',
+    'Mobile/SalesLogix/Models/QuickFormControls/CurrencyControl'
+
+], function(
+    declare,
+    lang,
+    _ModelBase,
+    TextControl,
+    AddressControl,
+    EmailControl,
+    PhoneControl,
+    NameControl,
+    DateControl,
+    BooleanControl,
+    UrlControl,
+    OwnerControl,
+    UserControl,
+    PicklistControl,
+    LookupControl,
+    NumericControl,
+    CurrencyControl
+) {
+
+    return declare('Mobile.SalesLogix.Models.QuickFormModel', [_ModelBase], {
+        
+        
+        /**
+         * @property {String}
+         * The unique (within the current form) name of the model
+         */
+        name: 'QuickFormModel',
+        dispalyName: 'Quick Form Model',
+        displayNamePlural: 'Quick Form Models',
+        modelData: {},
+        entityType: null,
+        resourceKind: null,
+        select: [],
+        include: [],
+        layout: [],
+        controlMaps:[
+            { name: 'text', type: 'Sage.Platform.QuickForms.Controls.QFTextBox, Sage.Platform.QuickForms', ctor: TextControl},
```

Comment:
Is there any reason we can't register the controls like we do with our fieldmanager? Then we could move the "type" to the control itself, and iterate the control collection for that type/name.